### PR TITLE
perf(dropdown): get option value judgment logic

### LINKED ISSUE #7064: Dropdown: onChange listener is not consistent in the event.value which is getting returned
### Describe the bug

event.value is not consistent when there is empty value selected with the dropdown.
The value getting returned is an object instead of a string

### Reproducer

https://stackblitz.com/edit/vitejs-vite-jemedx?file=src%2FApp.tsx

### System Information

  System:
    OS: Linux 5.0 undefined
    CPU: (6) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.20.3 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm
    pnpm: 8.15.6 - /usr/local/bin/pnpm
  npmPackages:
    primereact: 10.8.2 => 10.8.1 
    react: ^18.2.0 => 18.2.0 

### Steps to reproduce the behavior

1. Click on the dropdown item "All".
2. Observe the e.value returns as {name:"All",value:""}
3. Click on the dropdown item "Active".
4. Observe the e.value returns as "active".

### Expected behavior

The e.value when the value is empty should return empty value (i.e, "" )

### LINKED ISSUE #7081: Dropdown: null and empty string value
### Describe the bug

Dropdown select returns whole item object when value is null or empty string. Strangely it behaves correctly when val === false or 0

### Reproducer

https://stackblitz.com/edit/4db3vn?file=src%2FApp.jsx

### System Information

primereact 10.8.2

### Steps to reproduce the behavior

_No response_

### Expected behavior

_No response_

### PR DESCRIPTION
**Pull request by betavs**
### Defect Fixes
When submitting a PR, please also <ins>**create an issue**</ins> documenting the error and [manually link to an issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#manually-linking-a-pull-request-or-branch-to-an-issue-using-the-issue-sidebar) or mention it in the description using #<issue_id>.

### Feature Requests
Due to company policy, we are unable to accept feature request PRs with significant changes as such cases has to be implemented by our team following our own processes.
Smaller scaled feature implementations such as adding a property to a component will be considered for merging.

close #7064 
Fix #7081

### CODE REVIEW

**Comment by mtsh1000** [comment]
> ### Defect Fixes
> 
> When submitting a PR, please also **create an issue** documenting the error and [manually link to an issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#manually-linking-a-pull-request-or-branch-to-an-issue-using-the-issue-sidebar) or mention it in the description using #<issue_id>.
> ### Feature Requests
> 
> Due to company policy, we are unable to accept feature request PRs with significant changes as such cases has to be implemented by our team following our own processes. Smaller scaled feature implementations such as adding a property to a component will be considered for merging.
> 
> close #7064

This won't fix the issue, optionValue should always be returned if it was defined in props even if its null or ""

**Comment by melloware** [comment]
@mtsh1000 is correct this doesn't fix it.  however what if the original optionValue was also NULL or "" how would the component know that is what you are using?

File: components/lib/dropdown/Dropdown.js [hunk 1]
```diff
@@ -827,7 +827,7 @@ export const Dropdown = React.memo(
         const getOptionValue = (option) => {
             const optionValue = props.optionValue ? ObjectUtils.resolveFieldData(option, props.optionValue) : option ? option['value'] : ObjectUtils.resolveFieldData(option, 'value');
 
-            return ObjectUtils.isNotEmpty(optionValue) ? optionValue : option;
+            return optionValue ?? option;
```
Comment (melloware) [inline components/lib/dropdown/Dropdown.js:830]:
```suggestion
            return props.optionValue ? optionValue : option;
```

**Review by melloware — CHANGES_REQUESTED** [review]

File: components/lib/dropdown/Dropdown.js [hunk 2]
```diff
@@ -827,7 +827,7 @@ export const Dropdown = React.memo(
         const getOptionValue = (option) => {
             const optionValue = props.optionValue ? ObjectUtils.resolveFieldData(option, props.optionValue) : option ? option['value'] : ObjectUtils.resolveFieldData(option, 'value');
 
-            return ObjectUtils.isNotEmpty(optionValue) ? optionValue : option;
+            return props.optionValue ? optionValue : option;
```
Comment (melloware) [inline components/lib/dropdown/Dropdown.js:830]:
```suggestion
            return props.optionValue !== undefined ? props.optionValue : option;
```

**Review by melloware — CHANGES_REQUESTED** [review]

**Comment by melloware** [comment]
Actually I changed it to `!== undefined` because `null and ""` would be treated as falsy.

**Comment by mtsh1000** [comment]
> Actually I changed it to `!== undefined` because `null and ""` would be treated as falsy.

Great, thats the proper fix, thank you:)

File: components/lib/dropdown/Dropdown.js [hunk 3]
```diff
@@ -827,7 +827,7 @@ export const Dropdown = React.memo(
         const getOptionValue = (option) => {
             const optionValue = props.optionValue ? ObjectUtils.resolveFieldData(option, props.optionValue) : option ? option['value'] : ObjectUtils.resolveFieldData(option, 'value');
 
-            return ObjectUtils.isNotEmpty(optionValue) ? optionValue : option;
+            return props.optionValue !== undefined ? optionValue : option;
```
Comment (melloware) [inline components/lib/dropdown/Dropdown.js:830]:
```suggestion
            return optionValue !== undefined ? optionValue : option;
```

**Review by melloware — CHANGES_REQUESTED** [review]

File: components/lib/dropdown/Dropdown.js (hunk 3 above)
Comment (mtsh1000) [reply components/lib/dropdown/Dropdown.js]: props.optionValue should also be checked for undefined in line before:)

File: components/lib/dropdown/Dropdown.js [hunk 4]
```diff
@@ -827,7 +827,7 @@ export const Dropdown = React.memo(
         const getOptionValue = (option) => {
             const optionValue = props.optionValue ? ObjectUtils.resolveFieldData(option, props.optionValue) : option ? option['value'] : ObjectUtils.resolveFieldData(option, 'value');
 
-            return ObjectUtils.isNotEmpty(optionValue) ? optionValue : option;
+            return optionValue !== undefined ? optionValue : option;
```
Comment (melloware) [inline components/lib/dropdown/Dropdown.js:830]:
```suggestion
            return (props.optionValue ||  ObjectUtils.isNotEmpty(optionValue)) ? optionValue : option;
```

**Review by melloware — CHANGES_REQUESTED** [review]

**Comment by melloware** [comment]
I think now i actually have the best correct fix.  If either `props.optionValue` is set or the value is not empty it will use `optionValue`

File: components/lib/dropdown/Dropdown.js [hunk 5]
```diff
@@ -827,7 +827,7 @@ export const Dropdown = React.memo(
         const getOptionValue = (option) => {
             const optionValue = props.optionValue ? ObjectUtils.resolveFieldData(option, props.optionValue) : option ? option['value'] : ObjectUtils.resolveFieldData(option, 'value');
 
-            return ObjectUtils.isNotEmpty(optionValue) ? optionValue : option;
+            return (props.optionValue ||  ObjectUtils.isNotEmpty(optionValue)) ? optionValue : option;
```
Comment (melloware) [inline components/lib/dropdown/Dropdown.js:830]:
```suggestion
            return (props.optionValue || ObjectUtils.isNotEmpty(optionValue)) ? optionValue : option;
```

**Review by melloware — CHANGES_REQUESTED** [review]

File: components/lib/dropdown/Dropdown.js (hunk 5 above)
Comment (mtsh1000) [reply components/lib/dropdown/Dropdown.js]: What about:
        
const getOptionValue = (option) => {
            return props.optionValue !== undefined ? ObjectUtils.resolveFieldData(option, props.optionValue) : option ? option['value'] : ObjectUtils.resolveFieldData(option, 'value');
         }

**Pull request merged**